src/libraries/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/HashProviderDispenser.Unix.cs

```diff
@@ -44,22 +44,15 @@ internal static class OneShotHashProvider
 
                 int length = destination.Length;
 
-                fixed (byte* pKey = key)
-                fixed (byte* pSource = source)
-                fixed (byte* pDestination = destination)
-                {
-                    Check(Interop.Crypto.HmacOneShot(
-                        evpType,
-                        pKey,
-                        key.Length,
-                        pSource,
-                        source.Length,
-                        pDestination,
-                        ref length));
-                    Debug.Assert(length == hashSize);
-                }
+                Check(Interop.Crypto.HmacOneShot(
```

Comment:
Can we move the error code checking into the interop method?  Then change the `out written` into `return written`?  It's both a clarity and a correctness thing... if the Interop layer traps the errors then there's nowhere that we forget to check for them.